### PR TITLE
chore: Use Python `.gitignore` template

### PR DESCRIPTION
**Pull request by fpgmaas**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,166 @@
...
+# pytype static type analyzer
+.pytype/
 
-# VSCode
-.vscode
+# Cython debug symbols
+cython_debug/
 
-# Emacs
-.dir-locals.el
```
Comment (SemyonSinchenko) [inline .gitignore:23]: Why is it deleted? It is a file that allows to set up project-specific things. It is like `.vscode/config.json` file but for emacs. I'm using it.

File: .gitignore [hunk 2]
```diff
@@ -1,23 +1,166 @@
...
+# Pyre type checker
+.pyre/
 
-# MKDocs
-site
+# pytype static type analyzer
+.pytype/
 
-# VSCode
-.vscode
```
Comment (SemyonSinchenko) [inline .gitignore:20]: I would like to have it for the same reason like .dir-locals.el

File: .gitignore [hunk 3]
```diff
@@ -1,23 +1,166 @@
...
 
-# Emacs
-.dir-locals.el
\ No newline at end of file
+# PyCharm
+#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+#.idea/
```
Comment (SemyonSinchenko) [inline .gitignore:166]: Why is the row commented?

File: .gitignore [hunk 4]
```diff
@@ -1,23 +1,166 @@
...
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
```
Comment (SemyonSinchenko) [inline .gitignore:89]: Let's remove all these commented lines. I see zero reasons to have them.

File: .gitignore [hunk 5]
```diff
@@ -1,23 +1,166 @@
...
+# mypy
+.mypy_cache/
+.dmypy.json
+dmypy.json
+
+# Pyre type checker
+.pyre/
 
-# MKDocs
-site
```
Comment (SemyonSinchenko) [inline .gitignore:17]: I would like to have it. It may simplify the life of not very experienced contributors / students.

**Review by SemyonSinchenko — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (fpgmaas) [reply .gitignore:23]: Sorry, I thought this was included in the template. Added back!

File: .gitignore (hunk 2 above)
Comment (fpgmaas) [reply .gitignore:20]: Sorry, I thought this was included in the template. Added back!

File: .gitignore (hunk 5 above)
Comment (fpgmaas) [reply .gitignore:17]: it's added as `/site` by the template.

File: .gitignore (hunk 3 above)
Comment (fpgmaas) [reply .gitignore]: removed this section.

**Comment by fpgmaas** [comment]
@SemyonSinchenko sorry, I was probably a bit too fast with this PR :) Thanks for checking it; I updated it according to your feedback.

**Review by SemyonSinchenko — APPROVED** [review]
LGTM

**Pull request merged**